lib/src/source/git.dart

```diff
@@ -426,6 +433,20 @@ class BoundGitSource extends CachedSource {
     _updatedRepos.add(path);
   }
 
+  ///  Clean-up [dirPath] if it's an invalid git repository
+  void _cleanInvalidGitRepoCache(String dirPath) {
+    final Directory directory = Directory(dirPath);
+    if (dirExists(dirPath)) {
+      var processResult = runProcessSync(
+          git.command, ['rev-parse', '--is-inside-git-dir'],
+          workingDir: directory.path);
+      var result = processResult.stdout?.join('\n');
```

Comment:
I would suggest reading the exit code instead, it feels more robust... If it's zero then it's a git repository.